python/instrumentation/openinference-instrumentation-openai/src/openinference/instrumentation/openai/_utils.py

```diff
@@ -0,0 +1,134 @@
+import json
+import logging
+import warnings
+from enum import Enum
+from importlib.metadata import version
+from typing import (
+    Any,
+    Iterator,
+    List,
+    Mapping,
+    NamedTuple,
+    Optional,
+    Protocol,
+    Sequence,
+    Tuple,
+    Union,
+    cast,
+)
+
+from openinference.instrumentation.openai._with_span import _WithSpan
+from openinference.semconv.trace import SpanAttributes
+from opentelemetry import trace as trace_api
+from opentelemetry.util.types import Attributes, AttributeValue
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+_OPENAI_VERSION = tuple(map(int, version("openai").split(".")[:3]))
+
+
+class _MimeType(Enum):
+    text_plain = "text/plain"
+    application_json = "application/json"
+
+
+class _ValueAndType(NamedTuple):
+    value: str
+    type: _MimeType
```

Comment:
this will also end up being duplicated across so I'd suggest just centralizing into semantic-conventions as these value types need to be as semantic as the keys themselves.

Reply:
ok. that makes sense.